tools/lsp/properties.rs

```diff
@@ -187,7 +187,14 @@ fn get_properties(element: &ElementRc) -> Vec<PropertyInformation> {
         }
     }
 
-    result.sort_by(|a, b| a.name.cmp(&b.name));
+    result.extend(get_reserved_properties()); // Add reserved properties last!
+
+    // We can have duplicate properties that were defined and then changed further up the
+    // Element tree. So we need to remove duplicates.
+    result.sort_by(|a, b| a.name.cmp(&b.name)); // Sort is stable!
+    result.reverse(); // Now the deepest property is first
```

Comment:
I don't understand why you reverse

Reply:
I go from the actual type of the element down to the base types, adding properties for the current type.

The property is defined by the last entry in that list and gets reassigned values in the earlier versions. I do want the place it is defined at.

Sort is stable, so the definition still happens in the last occurrence of any property.

So I revert the list, moving the definition to the front, where dedup() will leave the definition in place and delete the other occurrences.

Then I reverse back to get the sorting back.

O tried using dedup from itertools, but that does not return a DoupleEndedIterator, so I can not reverse the list based on iterators after dedup.

Having thought about this some more: I think I can avoid one revert: I could sort in reverse order, reverse the Vec and then dedup that.

Reply:
I see, thanks for the explanation.